doc/Requirements Specification/Parts/Discussion.lyx

```diff
@@ -0,0 +1,283 @@
+\html_math_output 0
+\html_css_as_file 0
+\html_be_strict false
+\end_header
+
+\begin_body
+
+\begin_layout Chapter
+Discussion
+\end_layout
+
+\begin_layout Section
+Assumptions
+\end_layout
+
+\begin_layout Labeling
+\labelwidthstring 00.00.0000
+\begin_inset CommandInset label
+LatexCommand label
+name "/C10/"
+
+\end_inset
+
+ The measured software was built using 
+\begin_inset ERT
+status open
+
+\begin_layout Plain Layout
+
+
+\backslash
+gls{component-based software architecture}
+\end_layout
+
+\end_inset
+
+.
+ This assumption is derived from working with 
+\begin_inset ERT
+status open
+
+\begin_layout Plain Layout
+
+
+\backslash
+gls{Palladio}
+\end_layout
+
+\end_inset
+
+, which was built for analysing 
+\begin_inset ERT
+status open
+
+\begin_layout Plain Layout
+
+
+\backslash
+gls{component-based software}
+\end_layout
+
+\end_inset
+
+.
+ Fortunately, it most of the time imposes little loss of generality, as
+ any object oriented software can be described using terms of 
+\begin_inset ERT
+status open
+
+\begin_layout Plain Layout
+
+
+\backslash
+gls{component-based software architecture}
+\end_layout
+
+\end_inset
+
+ (regarding each class as a 
+\begin_inset ERT
+status open
+
+\begin_layout Plain Layout
+
+
+\backslash
+gls{component}
+\end_layout
+
+\end_inset
+
+ in the worst case).
+ Such software will naturally not have the advantages that come with the
+ 
+\begin_inset ERT
+status open
+
+\begin_layout Plain Layout
+
+
+\backslash
+gls{component-based software}
+\end_layout
+
+\end_inset
+
+ approach, but might still be analysed for their performance.
+\begin_inset Note Note
+status open
+
+\begin_layout Plain Layout
+I (Joshua) think this to actually be true, but fear that the SDQ won't agree
+ on this.
+\end_layout
+
+\end_inset
+
+
+\end_layout
+
+\begin_layout Labeling
+\labelwidthstring 00.00.0000
+\begin_inset CommandInset label
+LatexCommand label
+name "/C20/"
+
+\end_inset
+
+ The measured software has a constant, deterministic runtime for a fixed
+ configuration of input parameters, when ignoring influences of the hardware,
+ operating system and error of measurement.
+ This will be the case for most software.
+ The fact the user tries to measure the software when using Beagle implies
+ he expects it to behave in such a manner.
+\end_layout
+
+\begin_layout Section
+Challenges
+\end_layout
+
+\begin_layout Labeling
+\labelwidthstring 00.00.0000
+\begin_inset CommandInset label
+LatexCommand label
+name "/C100/"
+
+\end_inset
+
+ For accurate measurements, the CPU of the test server needs to be controlled.
+ This involves disabling turbo boost, reading the temperature of each core
+ and making sure the CPU is in a real world application thermal state, and
+ possibly further measures.
+\end_layout
+
+\begin_layout Labeling
+\labelwidthstring 00.00.0000
+\begin_inset CommandInset label
+LatexCommand label
+name "/C110/"
+
+\end_inset
+
+ Scalability and transferability need to be ensured.
```

Comment:
of Beagle? Then I would rather put this to /Q#/ and make it more precise (like: Beagle should be able to analyse a Desktop application on an average Desktop PC...)